### PR TITLE
Migrated SmartContractService to new design

### PR DESCRIPTION
**Pull request by netopyr**
This PR migrates the SmartContractService to the new design:

- Makes the `preHandle()` method generic. It has only one parameter `context` and is defined in `TransactionHandler`.
- Introduces interfaces for each store (e.g. `ReadableTokenStore`) in the corresponding api-module
- Adds the suffix 'Impl' to names of store implementations (e.g. `ReadableTokenStoreImpl`). The classes implement the new interfaces.
- `PreHandleContext` will become an interface in an upcoming PR. To prepare this change, `FakePreHandleContext` was introduced and tests were updated. This class allows to do the switch without breaking all the tests in services.
- We will stop to use `AccountAccess` soon. Instead the new interface `ReadableAccountStore` should be used instead.

### CODE REVIEW

**Review by hendrikebbers — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Review by jeromy-cannon — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**
